language/internal/cc/parser/parser_test.go

```diff
@@ -661,7 +661,48 @@ func TestParseConditionalIncludes(t *testing.T) {
 			},
 			expectedErrors: []string{
 				"3:11: expected integer literal or identifier, got newline",
-				"4:4: unpaired `#if`: directive '#endif'",
+			},
+		},
+		{
+			// Block starts with an invalid branch type
+			input: `
+			#else
+			#   include "should_not_parse.h"
+			#endif
+			`,
+			expected: nil,
+			expectedErrors: []string{
+				"2:4: unpaired `#if`: directive '#else'",
+			},
+		},
+		{
+			// Recover from an invalid expression
+			input: `
+			#if 128 >>
+			#include "dropped1.h"
+			#endif
+
+			#if defined(VALID)
+			#   if wrong &&
+			#       include "dropped2.h"
+			#   endif
+			#   include "reachable.h"
+			#endif
```

Comment:
Potentially we can add 1 more test case here: unterminated #if block, so #if without matching #endif. Seems like it should handle that already